modules/subscriber/variables.tf

```diff
@@ -0,0 +1,82 @@
+variable "name" {
+  type        = string
+  description = <<-EOF
+    Name of role. Since this name must be unique within the
+    account, it will be reused for most of the resources created by this
+    module.
+  EOF
+
+  validation {
+    condition     = length(var.name) <= 64
+    error_message = "Name must be under 64 characters."
+  }
+}
+
+variable "destination_iam_arn" {
+  description = "ARN for destination iam policy"
+  type        = string
+}
+
+variable "firehose_arn" {
+  description = "ARN for kinesis firehose"
+  type        = string
+}
+
+variable "filter_name" {
+  description = "Subscription filter name. Existing filters that have this name as a pre  fix will be removed."
+  type        = string
+}
+
+variable "log_group_name_patterns" {
+  description = "List of patterns as strings. We will only subscribe to log groups that have names matching one of the provided strings based on strings based on a case-sensitive substring search. To subscribe to all log groups, use the wildcard operator *."
+  type        = list(string)
```

Comment:
By omitting an argument you are making this field mandatory.

Both `log_group_name_patterns` and `log_group_name_prefixes` should have a default of `[]`, and not be nullable.